src/MIEApi.js

```diff
@@ -0,0 +1,138 @@
+import axios from 'axios';
+import logger from './logger.js'; 
+import { endpoints } from './apiConfig.js';
+
+class MIEApi {
+  static sessionCache = new Map();
+
+  constructor({ baseUrl, connectToken, userId, ip }) {
+    this.baseUrl = baseUrl;
+    this.connectToken = connectToken;
+    this.userId = userId;
+    this.ip = ip;
+    this.isRefreshingSession = false;
+  }
+
+  static getSessionKey(baseUrl, userId) {
+    return `${baseUrl}_${userId}`;
+  }
+
+  async refreshSession() {
+    if (this.isRefreshingSession) return;
+    this.isRefreshingSession = true;
+
+    try {
+      const refreshUrl = `${this.baseUrl}?f=layoutnouser&name=BlueHive_Refresh_Session&user_id=${encodeURIComponent(this.userId)}&connectToken=${encodeURIComponent(this.connectToken)}&ip_address=${encodeURIComponent(this.ip)}&raw&json`;
+
+      logger.info(`Refreshing connection to: ${refreshUrl}, 'for userId:', ${this.userId}`);
+
+      const response = await axios.get(refreshUrl, {
+        headers: {
+          'User-Agent': 'BlueHive AI (Refresh Connection)',
+        }
+      });
+
+      if (!response || response.data.status !== 200) {
+        throw new Error(response?.data?.message || 'Error connecting to WebChart.');
+      }
+
+      const getCookieResponse = await axios.get(`${this.baseUrl}?f=wcrelease&json`, {
+        headers: {
+          'User-Agent': 'BlueHive AI (Get x-db_name)',
```

Comment:
Do we need this?

Reply:
We have a default of mieapi but allow for an override

Reply:
@abroa01 any progress on this?